src/OmniSharp/Workers/Refactoring/FixUsings.cs

```diff
@@ -0,0 +1,238 @@
+                            }
+                        }
+
+                        if (usingOperations.Count() == 1)
+                        {
+                            //Only one operation - apply it
+                            usingOperations.Single().Value.Apply(workspace, CancellationToken.None);
+                            updated = true;
+                        }
+                        else if (usingOperations.Count() > 1)
+                        {
+                            if (usingOperations.Keys.Any(k => k.StartsWith("using Microsoft.CodeAnalysis.CSharp")))
+                            {
+                                //Resolve between charp/visualbasic
+                                var csharpOperation = usingOperations.Keys.Where(k => k.StartsWith("using Microsoft.CodeAnalysis.CSharp")).First();
+                                usingOperations[csharpOperation].Apply(workspace, CancellationToken.None);
+                                updated = true;
+                            }
+                            else
+                            {
+                                //What is left is ambiguous
+                                ambiguousNodes.Add(node);
+                                var unresolvedText = node.Identifier.ValueText;
+                                var unresolvedLocation = node.GetLocation().GetLineSpan().StartLinePosition;
+                                ambiguous.Add(new QuickFix
+                                {
+                                    Line = unresolvedLocation.Line + 1,
+                                    Column = unresolvedLocation.Character + 1,
+                                    FileName = fileName,
+                                    Text = "`" + unresolvedText + "`" + " is ambiguous"
+                                });
+                            }
+                        }
+                    }
+                }
+
+                processMore = updated;
+            }
+
+            return ambiguous;
+        }
+
+        private static async Task RemoveDuplicateUsings(string fileName, OmnisharpWorkspace workspace, Document document, SemanticModel semanticModel, ICodeActionProvider codeActionProvider)
+        {
+
+            var syntaxNode = (await document.GetSyntaxTreeAsync()).GetRoot();
+            var nodes = syntaxNode.DescendantNodes()
+                .Where(x => x is UsingDirectiveSyntax);
+
+            foreach (var node in nodes)
+            {
+                document = workspace.GetDocument(fileName);
+                semanticModel = await document.GetSemanticModelAsync();
+                var sourceText = (await document.GetTextAsync());
+                var diagnostics = semanticModel.GetDiagnostics();
+                var location = node.Span;
+                var actions = new List<CodeAction>();
+                var pointDiagnostics = diagnostics.Where(d => d.Location.SourceSpan.Contains(location)).ToImmutableArray();
```

Comment:
i.e. only look for `Unnecessary using directive.`. I think diagnostics have an ID that you could use.